Terraria/NPC.cs

```diff
@@ -84978,85 +85046,105 @@ public void NPCLoot()
 			}
 			IL_C1:
 			int num3 = Item.NPCtoBanner(this.BannerID());
-			if (num3 > 0 && !NPCID.Sets.ExcludedFromDeathTally[this.type] && this.AnyInteractions())
+			if (num3 > 0 && !NPCID.Sets.ExcludedFromDeathTally[this.type])
 			{
-				NPC.killCount[num3]++;
-				if (Main.netMode == 2)
+				bool flag2;
+				if (this.realLife >= 0)
+				{
+					flag2 = Main.npc[this.realLife].AnyInteractions();
+				}
+				else
 				{
-					NetMessage.SendData(83, -1, -1, "", num3, 0f, 0f, 0f, 0, 0, 0);
+					flag2 = this.AnyInteractions();
 				}
-				int num4 = 50;
-				if (NPC.killCount[num3] % num4 == 0 && num3 > 0)
+				if (flag2)
 				{
-					int l = Item.BannerToNPC(num3);
-					NPC nPC = new NPC();
-					nPC.netDefaults(l);
-					int num5 = this.lastInteraction;
-					if (!Main.player[num5].active || Main.player[num5].dead)
+					NPC.killCount[num3]++;
+					if (Main.netMode == 2)
 					{
-						num5 = this.FindClosestPlayer();
+						NetMessage.SendData(83, -1, -1, "", num3, 0f, 0f, 0f, 0, 0, 0);
 					}
-					string text = string.Concat(new object[]
-					{
-						"The ",
-						NPC.killCount[num3],
-						"th ",
-						Lang.npcName(l, false),
-						" has been defeated!"
-					});
-					if (num5 >= 0 && num5 < 255)
+					int num4 = 50;
+					if (NPC.killCount[num3] % num4 == 0 && num3 > 0)
 					{
-						text = string.Concat(new object[]
+						int l = Item.BannerToNPC(num3);
+						NPC nPC = new NPC();
+						nPC.netDefaults(l);
+						int num5 = this.lastInteraction;
+						if (!Main.player[num5].active || Main.player[num5].dead)
+						{
+							num5 = this.FindClosestPlayer();
+						}
+						string text = string.Concat(new object[]
 						{
-							Main.player[num5].name,
-							" has defeated the ",
+							"The ",
 							NPC.killCount[num3],
 							"th ",
 							Lang.npcName(l, false),
-							"!"
+							" has been defeated!"
 						});
+						if (num5 >= 0 && num5 < 255)
+						{
+							text = string.Concat(new object[]
+							{
+								Main.player[num5].name,
+								" has defeated the ",
+								NPC.killCount[num3],
+								"th ",
+								Lang.npcName(l, false),
+								"!"
+							});
+						}
+						if (Main.netMode == 0)
```

Comment:
If you look below that was already in the source, I did not add it. the only lines that changed are 85051-85061